File: app/src/main/java/com/example/coursecomrade/ProfessorAdapter.java
package com.example.coursecomrade;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class ProfessorAdapter extends RecyclerView.Adapter<ProfessorAdapter.ProfessorViewHolder> {

    private final Context context;
    private final List<ProfessorData> professorList;

    // Constructor
    public ProfessorAdapter(Context context, List<ProfessorData> professorList) {
        this.context = context;
        this.professorList = professorList;
    }

    // ViewHolder class
    public static class ProfessorViewHolder extends RecyclerView.ViewHolder {
        public final TextView name;
        public final TextView rating;

        public ProfessorViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.professorName);
            rating = itemView.findViewById(R.id.professorRating);
        }
    }

    @NonNull
    @Override
    public ProfessorViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.professor_item, parent, false);
        return new ProfessorViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ProfessorViewHolder holder, int position) {
        ProfessorData professor = professorList.get(position);

        // Set professor name
        holder.name.setText(professor.getFirstName() + " " + professor.getLastName());

        // Check if the rating is 0
        if ("0".equals(professor.getRating())) {
            holder.rating.setText("Not yet rated");
        } else {
            holder.rating.setText(professor.getRating() + " Stars");
        }

        // Set alternating background colors
        int backgroundColor;
        switch (position % 3) {
            case 0:
                backgroundColor = R.color.tan_shade_1;
                break;
            case 1:
                backgroundColor = R.color.tan_shade_2;
                break;
            default:
                backgroundColor = R.color.tan_shade_3;
                break;
        }
        holder.itemView.setBackgroundResource(backgroundColor);

        // Handle click to open ProfessorDetailActivity
        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(context, ProfessorDetailActivity.class);
            intent.putExtra("PROFESSOR_ID", professor.getProfessorId());
            intent.putExtra("PROFESSOR_NAME", professor.getFirstName() + " " + professor.getLastName());
            intent.putExtra("PROFESSOR_EMAIL", professor.getEmail());
            intent.putExtra("PROFESSOR_RATING", "0".equals(professor.getRating()) ? "No Rating Available" : professor.getRating());
            context.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return professorList.size();
    }
}

File: app/src/main/java/com/example/coursecomrade/ProfessorAdapter.kt
package com.example.coursecomrade

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.coursecomrade.R

class ProfessorAdapter(
    private val context: Context,
    private val professorList: List<ProfessorData>
) : RecyclerView.Adapter<ProfessorAdapter.ProfessorViewHolder>() {

    class ProfessorViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val name: TextView = itemView.findViewById(R.id.professorName)
        val rating: TextView = itemView.findViewById(R.id.professorRating)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProfessorViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.professor_item, parent, false)
        return ProfessorViewHolder(view)
    }

    override fun onBindViewHolder(holder: ProfessorViewHolder, position: Int) {
        val professor = professorList[position]

        // Set professor name
        holder.name.text = "${professor.firstName} ${professor.lastName}"

        // Check if the rating is 0
        holder.rating.text = if (professor.rating == "0") {
            "Not yet rated"
        } else {
            "${professor.rating} Stars"
        }

        // Set alternating background colors
        val backgroundColor = when (position % 3) {
            0 -> R.color.tan_shade_1
            1 -> R.color.tan_shade_2
            else -> R.color.tan_shade_3
        }
        holder.itemView.setBackgroundResource(backgroundColor)

        // Handle click to open ProfessorDetailActivity
        holder.itemView.setOnClickListener {
            val intent = Intent(context, ProfessorDetailActivity::class.java)
            intent.putExtra("PROFESSOR_ID", professor.professorId)
            intent.putExtra("PROFESSOR_NAME", "${professor.firstName} ${professor.lastName}")
            intent.putExtra("PROFESSOR_EMAIL", professor.email)
            intent.putExtra("PROFESSOR_RATING", if (professor.rating == "0") "No Rating Available" else professor.rating)
            context.startActivity(intent)
        }
    }


    override fun getItemCount() = professorList.size
}
